MoreLinq.Test/DuplicatesTest.cs

```diff
@@ -0,0 +1,140 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2023 Julien Aspirot. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq.Test
+{
+    using System.Collections.Generic;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class DuplicatesTest
+    {
+        [Test]
+        public void Duplicates_IsLazy()
+        {
+            _ = new BreakingSequence<object>().Duplicates();
+        }
+
+        [Test]
+        public void Streams_Duplicates_As_They_Are_Discovered()
+        {
+            static IEnumerable<string> Source()
+            {
+                yield return "DUPLICATED_STRING";
+                yield return "DUPLICATED_STRING";
+                throw new TestException();
+            }
+
+            using var source = TestingSequence.Of(Source());
+
+            void Act() => source.Duplicates().Take(1);
+
+            Assert.That(Act, Throws.Nothing);
```

Comment:
This test doesn't work at all. I didn't mean you to copy [the test code I suggested verbatim](https://github.com/morelinq/MoreLINQ/pull/1037#discussion_r1396001596), but use it as a starting point and guidance because it was written from the head and clearly flawed. 😅 Looking at this now, `Duplicates` and `Take` are lazy and their resulting sequences never iterated, so no exception will ever get thrown and the test will always pass. In fact, in you set a breakpoint at the start of `Sources`, you'll find it's never called.

I suggest fixing this by asserting the result so we know work was done (and that if the test passes then it never got as far as throwing the exception):

```suggestion
            using var source = Source().AsTestingSequence();

            var duplicates = source.Duplicates().Take(1).ToArray();

            Assert.That(duplicates, Is.EqualTo(new[] { "DUPLICATED_STRING" }));
```

Reply:
perfect, i'll apply that. I've forgot about the lazy part in a couple of assert. I dont know why i always assume that the assert code will enumerate the collection.